app/main.py

```diff
@@ -1,3 +1,13 @@
 def cache(func):
-    # Write your code here
-    pass
+    dict_with_args = {}
+
+    def wrapper(*args):
+        if args in dict_with_args:
+            print("Getting from cache")
+            return dict_with_args[args]
+
+        dict_with_args.update({args: func(*args)})
+        print("Calculating new result")
+        return func(*args)
```

Comment:
Here you call the function a second time. The first time you call it in the 9th line.
It's better you use the already calculated value from the dict :)